cartography/rules/cli.py

```diff
@@ -0,0 +1,166 @@
+"""
+Cartography RunRules CLI
+
+Execute security frameworks and present facts about your environment.
+"""
+
+import argparse
+import getpass
+import logging
+import os
+import sys
+
+from cartography.rules.data.frameworks import FRAMEWORKS
+from cartography.rules.runners import run_frameworks
+
+
+class CLI:
+    """
+    Command line interface for Cartography security framework execution.
+    """
+
+    def __init__(self, prog=None):
+        self.prog = prog
+        self.parser = self._build_parser()
+
+    def _build_parser(self):
+        """
+        Build the argument parser for the rules CLI.
+
+        :rtype: argparse.ArgumentParser
+        :return: A rules argument parser.
+        """
+        parser = argparse.ArgumentParser(
+            prog=self.prog,
+            description="Execute Cartography security frameworks",
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            epilog="For more documentation please visit: https://github.com/cartography-cncf/cartography",
+        )
+
+        available_frameworks = list(FRAMEWORKS.keys()) + ["all"]
+        parser.add_argument(
+            "framework",
+            nargs="?",
+            choices=available_frameworks,
+            help='Security framework to execute (or "all" to execute all frameworks)',
+        )
+
+        parser.add_argument(
+            "--list", action="store_true", help="List available frameworks and exit"
+        )
+
+        parser.add_argument(
```

Comment:
It would be nice to eventually store these common arguments between CLIs in one place (like uri, username, password, etc). If the CLI args or the env vars got out of sync it would hinder usability. For example Neo4j tends to use the word "username" and as such Langgraph and others use the env var `NEO4J_USERNAME` whereas we use `NEO4J_USER`, it would be unfortunate for us to somehow use both USER and USERNAME across different CLIs

Reply:
Yeah, will do once we move the main CLI over to typer